lib/sendgrid_api/delivery_methods/sendgrid.rb

```diff
@@ -103,6 +104,22 @@ def parse_xsmtpapi_headers(mail)
     #
     def parse_email_address(email)
       Array(email).collect { |email| Mail::Address.new(email) }
+    rescue Mail::Field::ParseError => e
+      punycode_email(email)
     end
+
+    # convert email domain names containing unicode characters to punycode
+    # @param [String] email
+    # @return [Array] Mail::Address
+    #
+    def punycode_email(email)
+      Array(email).collect do |email|
+        email_part, display_name = email.split(/\<(.*?)\>/).reverse
+        email = Mail::Address.new(SimpleIDN.to_ascii(email_part))
+        email.tap{ |m| m.display_name = display_name.tr!('"','').strip!} if display_name
```

Comment:
With the pattern improvement, we won't need strip to remove whitespace separator.

Also we should avoid use of `tr` which could be problematic if there are quotes within the display name e.g. `Scott "Hurricane" Barrow`. (They get escaped with backslashes on output). Prefer something like: 

```
display_name[1..-2] if display_name[0] == '"'
```